input/kafka.go

```diff
@@ -88,22 +96,37 @@ func NewKafka(brokers []string, topic string, consumerGroup string, kafkaConfig
 	} else {
 		logger.Info("kafka input init correctly")
 	}
+
 	metrics.RegisterKafkaConsumerMetrics("sarama", kafkaConfig)
+
 	return &Kafka{
-		BaseInput: BaseInput{handler: h, name: fmt.Sprintf("kafka[topic=%s;cg=%s;id=%s]", topic, consumerGroup, kafkaConfig.ClientID)},
-		topic:     topic,
-		client:    client,
-		cg:        cg,
-		groupID:   consumerGroup,
-		ctx:       context.Background(),
-		closed:    make(chan bool),
-		logger:    logger,
+		BaseInput:  BaseInput{handler: h, name: fmt.Sprintf("kafka[topic=%s;cg=%s;id=%s]", topic, consumerGroup, kafkaConfig.ClientID)},
+		config:     kafkaConfig,
+		topic:      topic,
+		client:     client,
+		cg:         cg,
+		groupID:    consumerGroup,
+		ctx:        context.Background(),
+		closed:     make(chan bool),
+		logger:     logger,
 		enableTags: enableTags,
+		contexts:   make(map[int32]PartitionContext),
+	}
+}
+
+func (k *Kafka) initializePartitionContext(partition int32) {
+	k.contexts[partition] = PartitionContext{
+		lagGauge:       k.config.MetricRegistry.GetOrRegister(fmt.Sprintf("consumer_lag_%d", partition), gometrics.NewGauge()).(gometrics.Gauge),
```

Comment:
You could use prometheus library and deal with labels that might be simpler (like the others in carbon-relay-ng/metrics)